src/DocFxTocGenerator/DocFxTocGenerator/Index/IndexService.cs

```diff
@@ -0,0 +1,88 @@
+﻿// <copyright file="IndexService.cs" company="DocFx Companion Tools">
+// Copyright (c) DocFx Companion Tools. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+// </copyright>
+using DocFxTocGenerator.FileService;
+using DocFxTocGenerator.Liquid;
+using Microsoft.Extensions.Logging;
+
+namespace DocFxTocGenerator.Index;
+
+/// <summary>
+/// Service to generate an index for a folder.
+/// </summary>
+public class IndexService
+{
+    private readonly LiquidService _liquidService;
+    private readonly IFileService _fileService;
+    private readonly ILogger _logger;
+
+    private readonly string _defaultIndexTemplate =
```

Comment:
maybe add some comments on liquid template as not everyone is familiar with it

Reply:
Added some links in the LiquidService header. Other documentation is in the README.md.